src/main/scala/com/databricks/labs/overwatch/pipeline/Schema.scala

```diff
@@ -258,11 +284,87 @@ object Schema extends SparkSessionWrapper {
+    StructField("spark_submit_params",ArrayType(StringType, containsNull = true), nullable = true),
+    StructField("notebook_params",ArrayType(StringType, containsNull = true), nullable = true)
+  ))
+
+  val minimumSubmitRunDetailsSchema: StructType = StructType(Seq(
+    StructField("existing_cluster_id",StringType, nullable = true)
+  ))
+
+  val minimumRepairDetailsSchema: ArrayType = ArrayType(
+    StructType(Seq(
+      StructField("repair_timestamp",LongType, nullable = true),
+      StructField("rerun_tasks",StringType, nullable = true),
+      StructField("latest_repair_id",StringType, nullable = true)
+    ))
+  )
+
+  val minimumNotebookTaskSchema: StructType = StructType(Seq(
+    StructField("notebook_path",StringType, nullable = true),
+    StructField("revision_timestamp",LongType, nullable = true)
+  ))
+
+  val minimumSparkPythonTaskSchema: StructType = StructType(Seq(
+    StructField("spark_python_task",StructType(Seq(
+      StructField("parameters",ArrayType(StringType, containsNull = true), nullable = true),
+      StructField("python_file",StringType)
+    )),nullable = true)
+  ))
+
+  val minimumSparkSubmitTaskSchema: StructType = StructType(Seq(
+    StructField("parameters",ArrayType(StringType, containsNull = true), nullable = true)
+  ))
+
+  val minimumSparkJarTaskSchema: StructType = StructType(Seq(
+    StructField("jar_uri",StringType,nullable = true),
+    StructField("main_class_name",StringType,nullable = true),
+    StructField("parameters",ArrayType(StringType, containsNull = true), nullable = true)
+  ))
+
+  val minimumShellCommandTaskSchema: StructType = StructType(Seq(
+    StructField("command",ArrayType(StringType,containsNull = true),nullable = true)
+  ))
+
+  val minimumPipelineTaskSchema: StructType = StructType(Seq(StructField("pipeline_id",StringType,nullable = true)))
+
+  val minimumPythonWheelTaskSchema: StructType = StructType(Seq(
+    StructField("entry_point",StringType, nullable = true),
+    StructField("package_name",StringType, nullable = true),
+    StructField("parameters",ArrayType(StringType, containsNull = true), nullable = true)
+  ))
+
+  val minimumSQLTaskSchema: StructType = StructType(Seq(
+    StructField("query",StructType(Seq(
+      StructField("query_id",StringType, nullable = true))
+    ), nullable = true)
+  ))
+
+  val minimumTaskDetailSchema: StructType = StructType(Seq(
+    StructField("notebook_task",minimumNotebookTaskSchema, nullable = true),
+    StructField("spark_python_task",minimumSparkPythonTaskSchema, nullable = true),
+    StructField("python_wheel_task",minimumPythonWheelTaskSchema, nullable = true),
+    StructField("spark_jar_task",minimumSparkJarTaskSchema, nullable = true),
+    StructField("spark_submit_task",minimumSparkSubmitTaskSchema, nullable = true),
+    StructField("shell_command_task",minimumShellCommandTaskSchema, nullable = true),
+    StructField("pipeline_task",minimumPipelineTaskSchema, nullable = true),
```

Comment:
add dbt & sql tasks

Reply:
these may need to be handled in the conversion from struct to map but not necessary in the minSchema.